feature-libs/asm/components/asm-save-cart-dialog/asm-save-cart-dialog.component.ts

```diff
@@ -0,0 +1,69 @@
+/*
+ * SPDX-FileCopyrightText: 2023 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { Cart } from '@spartacus/cart/base/root';
+import { GlobalMessageType } from '@spartacus/core';
+import { FocusConfig, LaunchDialogService } from '@spartacus/storefront';
+import { Subscription } from 'rxjs';
+
+export enum SAVE_CART_DIALOG_ACTION {
+  CANCEL = 'CANCEL',
+  SAVE = 'SAVE',
+}
+
+@Component({
+  selector: 'cx-asm-save-cart-dialog',
+  templateUrl: './asm-save-cart-dialog.component.html',
+})
+export class AsmSaveCartDialogComponent implements OnInit, OnDestroy {
+  BIND_CART_ACTION = SAVE_CART_DIALOG_ACTION;
+  showDialogAlert = true;
+  globalMessageType = GlobalMessageType;
+  private subscription = new Subscription();
```

Comment:
In general, we prefer not to use `private` in our library because this makes the implementation invisible to users. Sometimes our lib users might like to customize a method. Methods we prefer users not to touch we use `protected` for as this still allows users to changes them if they really want to.

Reply:
fixed